spec/array_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'array'
+
+describe 'Array' do
+
+  let(:test) {Array.new}
+
+  def populate
+    test << 1
+    test << 2
+    test << 500
+    test << 30
+  end
+
+  describe 'Fully functional inject' do
+
+    it 'raises an error if not passed a block' do
+      populate
+      expect{test.my_inject}.to raise_error
+    end
+
+    it 'takes the block {|sum, x| sum + x} and return cumulative sum' do
+      populate
+      # [1,2,500,30]
```

Comment:
Comments aren't needed if your code/tests are descriptive enough - you can declare this in a `let`